cdap-watchdog/src/main/java/io/cdap/cdap/logging/clean/LogFolderCleaner.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.logging.clean;
+
+import org.apache.twill.filesystem.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * LogFolderCleaner is responsible for cleaning up accumulated empty logs folders.
+ */
+public class LogFolderCleaner implements Runnable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(LogFolderCleaner.class);
+
+  private final Location logsDirectoryLocation;
+  private final int maxFolderCount;
+
+  private int count;
+
+  /**
+   * Parameterized constructor for LogFolderCleaner.
+   *
+   * @param logsDirectoryLocation Logs parent folder location.
+   * @param maxFolderCount Maximum number of folders to be cleaned up
+   */
+  public LogFolderCleaner(Location logsDirectoryLocation, int maxFolderCount) {
+    this.logsDirectoryLocation = logsDirectoryLocation;
+    this.maxFolderCount = maxFolderCount;
+  }
+
+  @Override
+  public void run() {
+    try {
+      LOG.debug("Starting log folder cleanup at timestamp {}", System.currentTimeMillis());
+      cleanupEmptyFolders(logsDirectoryLocation);
+      LOG.debug("Completed cleaning up {} log folders at timestamp {}", count,
```

Comment:
nit: you usually don't need to put time here as it's in the logger line prefix already. It would be more useful to show the time elapsed

Reply:
Done